lib/path/default.nix

```diff
@@ -321,6 +344,62 @@ in /* No rec! Add dependencies on this file at the top. */ {
       subpath = joinRelPath deconstructed.components;
     };
 
+  /*
+    Whether a [path](https://nixos.org/manual/nix/stable/language/values.html#type-path)
+    has a [store path](https://nixos.org/manual/nix/stable/store/store-path.html#store-path)
+    as a prefix.
+
+    :::{.note}
+    As with all functions of this path library, it does not work on paths in strings,
+    which is how you'd typically get store paths.
+
+    Instead this function only handles path values themselves,
+    which occur when Nix files in the store use relative path expressions.
+    :::
+
+    Type:
+      hasStorePathPrefix :: Path -> Bool
+
+    Example:
```

Comment:
Reorder these to convey the most important information fastest.

1. `/nix/store/nvl9ic0pj1fpyln3zaqrf4cclbqdfn1j-foo/bar/baz`
2. `/nix/store`
3. `/nix/store/nvl9ic0pj1fpyln3zaqrf4cclbqdfn1j-foo`
4. the other things that aren't store paths.
5. the derivation (I think virtually everyone would guess that one right)

Reply:
There's `drvPath` on all packages, but the semantics are odd.
- https://github.com/NixOS/nix/issues/7910

There's little reason to reference derivation paths as strings, let alone path values, so this is more than good enough.